packages/react-components/src/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -3,17 +3,28 @@ import cx from 'clsx';
 
 import { ButtonSize, ButtonProps } from '../Button';
 
-import styles from './ButtonGroup.module.scss';
+import styles from './SegmentedControl.module.scss';
 
 import noop from '../../utils/noop';
 
-const baseClass = 'btn-group';
+const baseClass = 'segmented-control';
 
-export interface ButtonGroupProps extends React.HTMLAttributes<HTMLDivElement> {
+export type ButtonState =
+  | 'active'
+  | 'hover'
+  | 'enabled'
+  | 'disabled'
```

Comment:
Shouldn't we derive the state from the `Button` component?

Reply:
☝️ or move that state to some kind of const and use it here and in the `Button` component